### PR TITLE
Feature: Open all items in a tag

### LINKED ISSUE #7493: Feature: Option to open all items in a tag
### What's the Problem?

People usually use the same group of apps for common tasks. 

Right now Bundles are just a fancy quick-access bookmarks, but allowing to open all contained items at once (when the bundle is clicked) would make them much more useful

### Solution/Idea

Open all contained items at once when the Bundle is clicked

### Alternatives

Click+Shift+Click to select all items then Enter to open them

### Priorities

| Capability | Priority |
| :---------- | :------- |
| Allow users to still open items individually when the item name is clicked | Must |
| Allow users to open all items in 1 click, probably from a menu flyout on the top right of the tag container on the home page | Must |

### Files Version

.34.0

### Windows Version

21H2

### Comments

_No response_

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #7493

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open app and navigate to `Home`
   2. Enable the `Tags` widget
   3. Right click on the title bar
   4. Click on `Open all`

**Screenshots (optional)**
Screenshot 2023-07-18 224854

### CODE REVIEW

File: src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+namespace Files.App.Actions
+{
+    sealed class OpenAllTaggedActions: ObservableObject, IAction
+    {
+		private readonly IContentPageContext _pageContext;
+
+		private readonly ITagsContext _tagsContext;
+
+		public string Label
+			=> "OpenAllItems".GetLocalizedResource();
```
Comment (yaira2) [inline src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs:13]:
```suggestion
			=> "OpenAllTaggedItems".GetLocalizedResource();
```

**Comment by yaira2** [comment]
@ferrariofilippo I pushed a small design tweak so that clicking the tag name does the search for more items. This will allow us to repurpose the "view more" button to "open all".

**Comment by ferrariofilippo** [comment]
What about using our OpacityIcons? The Glyph looks strange
image

File: src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs (hunk 1 above)
Comment (QuaintMako) [inline src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs:13]: The label should be `OpenAllTaggedItems`

File: src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+		public string Description
+			=> "OpenAllItemsDescription".GetLocalizedResource();
```
Comment (QuaintMako) [inline src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs:16]: `OpenAllTaggedItemsDescription`

File: src/Files.App/Strings/en-US/Resources.resw [hunk 3]
```diff
@@ -3398,4 +3398,10 @@
   <data name="OpenInNewWindowDescription" xml:space="preserve">
     <value>Open directory in new window</value>
   </data>
+  <data name="OpenAllItems" xml:space="preserve">
+    <value>Open all</value>
+  </data>
+  <data name="OpenAllItemsDescription" xml:space="preserve">
+    <value>Open all tagged items</value>
+  </data>
```
Comment (QuaintMako) [inline src/Files.App/Strings/en-US/Resources.resw:3406]: `OpenAllTaggedItems` and `OpenAllTaggedItemsDescription`

File: src/Files.App/UserControls/Widgets/FileTagsWidget.xaml [hunk 4]
```diff
@@ -51,39 +52,44 @@
...
+							</StackPanel>
+						</HyperlinkButton>
+
+						<!--  Open All Items  -->
+						<HyperlinkButton
+							Grid.Column="2"
+							Margin="4"
+							AutomationProperties.Name="{helpers:ResourceString Name=OpenAllItems}"
+							Command="{x:Bind OpenAllCommand}"
+							ToolTipService.ToolTip="{helpers:ResourceString Name=OpenAllItems}">
```
Comment (QuaintMako) [inline src/Files.App/UserControls/Widgets/FileTagsWidget.xaml:90]: If the resources strings are indeed renamed as proposed, these entries will need to be renamed as well.

**Review by QuaintMako — COMMENTED** [review]
Code looks good, I am not able to test it.
A small proposition of renaming to stay coherent with the current.

File: src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs]: I thought this as well but then I realized this string might be useful outside of tags.

Comment (ferrariofilippo) [reply src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs]: Should I revert it then?

**Comment by ferrariofilippo** [comment]
> @ferrariofilippo I don't think the opacity icon is a good fit but to be honest, there isn't an intuitive icon for "open all". An alternative would be to use the `more` icon and display the menu item to "open all" when clicking the icon. 

Good call, we won't need to change the code if we will need any extra actions

**Comment by ferrariofilippo** [comment]
image
 Should we remove the button border?

**Comment by yaira2** [comment]
Can you set it to transparent?

**Comment by yaira2** [comment]
Does this also work for the sidebar?

**Comment by ferrariofilippo** [comment]
Not yet, I'll work on it

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**